mautic/mautic
# Populate form fields with the type number

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | yes
| New feature/enhancement? (use the a.x branch)      | yes
| Deprecations?                          | no
| BC breaks? (use the c.x branch)        | no
| Automated tests included?              | yes
| Related user documentation PR URL      | not necessary
| Related developer documentation PR URL |  N/A
| Issue(s) addressed                     | none opened

#### Description:
A form field that has a custom field of the type number mapped, is not prepopulated in the form. This PR adds or fixes the possibility to Auto fill this data.

#### Steps to test this PR:
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new form that has fields that map to contact or company fields with a type number. 
<img width="828" alt="image" src="https://github.com/mautic/mautic/assets/13075514/fc184a8c-469f-4bc6-b693-6d5f257f4be7">
4. Make sure the pre-fill option is enabled on the field <img width="736" alt="image" src="https://github.com/mautic/mautic/assets/13075514/ea071149-b85a-4fa8-9021-26856e80fff8">
5. Add the form to a landingpage
6. Get a trackable link to this landingpage (so the contact is identified). E.g. by sending a personal email with the link to the landingpage (it has to be a html link, the URL as string in the body of the email does not generate a trackable link)
7. Test it without being logged in with Mautic (e.g. in Incognito)
8. The form should be prefilled with all the values on the contact. Including the number.
